src/traces/bar/attributes.js

```diff
@@ -39,6 +39,51 @@ var marker = extendFlat({
         max: 1,
         editType: 'style',
         description: 'Sets the opacity of the bars.'
+    },
+    patternfill: {
+        shape: {
+            valType: 'enumerated',
+            values: ['', '/', '\\', 'x', '-', '|', '+', '.'],
+            dflt: '',
+            arrayOk: true,
+            editType: 'style',
+            description: [
+                'Sets the shape of the pattern fill.',
+                'By default, no pattern is used for filling the area.',
+            ].join(' ')
+        },
+        bgcolor: {
+            valType: 'color',
+            arrayOk: true,
+            editType: 'style',
+            description: [
+                'Sets the background color of the pattern fill.',
+                'Defaults to a transparent background.',
+            ].join(' ')
+        },
+        scale: {
+            valType: 'number',
+            min: 0,
+            dflt: 1,
+            arrayOk: true,
+            editType: 'style',
+            description: [
+                'Sets the scale of the pattern fill.',
+                'Defaults to 1.',
```

Comment:
Can we make this more meaningful? Looks like `1` means "8 px unit square" - can we change it to just `dflt: 8` and document it as the size of the unit square?

Also if `dflt` is specified you don't need to say in the description what the default is.

Reply:
Actually, the size of the unit square was also my first choice, but I changed it to the scaling factor so that stripes (`/`, `\`, `-`, `|`) look nice by default. I wanted to make intervals of stripes the same by default, but it was difficult when the size of unit square is fixed. Because of this reason, I set `baseSize * Math.sqrt(2)` to the size of unit square of `/`, `\`, and `x`.

But I agree that `scale` attribute is hard to interpret. Do you have any good idea?

Reply:
With the sqrt(2) factor we can still think of them all as unit squares of the same size, it's just that the unit square is rotated 45º for the diagonal patterns - the pattern you define, aligned vertically and horizontally, is then actually twice the fundamental repeating unit square: one copy in the middle, with corners at the midpoints of the edges of the pattern, and the other formed by taking the four leftover corner pieces and rearranging them back together into a square.

Reply:
You're right! I changed `scale` to `size`, which specifies the size of unit squares.
39e35ccc02eb5d76ea7bbdc88be61a7cb1943918